### DISCUSSION TITLE
Poll/Discussion to confirm current `SDL stereotype` naming

### DISCUSSION THREAD

**Discussion by The-Lum** [plantuml/plantuml#1270] [Polls]
Hi all,

According to [QA-11518](https://forum.plantuml.net/11518/issues-with-final-separator-latex-math-expression-activity?show=17284#c17284):
- Here is a poll to confirm current `SDL stereotype` naming:
>[![](https://www.plantuml.com/plantuml/svg/HP2nJWCn38RtF8MbBd2FaAjA9HWOs5YxM3rp5oMr8yUXEY7UdVgAOFB_VP-ovDWCrVgbX6s4wv8AmSCml5AfqDimn1ERm7aHQGIs46YQ5uDCwrLqEl67ABo_l44pKX9RAyN7CAlq2hi9DS-AAto5cBXs6s6_tyRX4A9q-rM-xgwgd6dgIgx_m4iHd5pwTMxuISv-dS_2bhXBRsx_oPjXosxztlWxj5gIXLt5GhCISPGyXcPIxmLmniItdZX75WFFyLbwcM2L3XVC10XLfKd3yZG6ucbxwMFxmx0JHPufie3Kn4bu3CTRxPVo0m00)](https://www.plantuml.com/plantuml/uml/HP11JiGm34NtEOKbDj0ZfADX93PSmsfC6oKJHx7DgBUdxY1O_V_UbpU-YU9GkzTmbl3OSYLudQPFgXrCfWcKOTcOXK0tWf7NJQ7G_k2Hu8i7YD8WrhrJV0injsuwm-LobkirH3RzLTwUhWzUADaWrt_WIsLCBZsT1R_9sTDvuQQv6Pku_IVV5AMuzJnOUiqQNZfMMWTHYrpc8ChzkG2kcDknvrPYOmMVumTRJR2pmHqB0K8VB2nOt-T0BPsNNiv7XTlHZkVz0000)

_Some `SDL` ref.:_
- http://sdl-forum.org/sdl88tutorial/3.ProcessDefinition/3.1_States_and_transitions.htm
- http://sdl-forum.org/sdl88tutorial/3.ProcessDefinition/3.7_Procedures.htm
- http://www.sdl-forum.org/Tools/TIMe/HTML/elb/sdl/mainsdl.htm
- https://www.graphviz.org/doc/info/shapes.html#sdlshapes
- https://www.ibm.com/docs/en/filenet-p8-platform/5.2.0?topic=objects-standard-microsoft-visio-shapes-organized-by-stencil#bpfwd046__SDL

Regards.

**Comment by The-Lum** [comment]
Hi PlantUML team, @arnaudroques,

It seems that `load` and `save` shape are reversed.
See:
- http://www.sdl-forum.org/Tools/TIMe/HTML/elb/sdl/sdl_z47.htm#top

Regards.

**Comment by fuhrmanator** [comment]
Hi team and @arnaudroques, 

I've been using "objects" in activity diagrams (not SDL), e.g., `:Passenger List]` which is the output of an activity. See https://sourcemaking.com/uml/modeling-for-system-integration/process-view/activity-diagrams

So it's awkward to have to call them `<<task>>` now (the semantics are not right). I suggest some solutions:

- A `skinparam` to say we're not doing SDL. My old diagrams are easier to fix this way. Currently many of my blogs now have the deprecation message:  
![image](https://github.com/user-attachments/assets/d201684e-8c1d-4f73-9540-977af6efa0b3)

- A synonym, e.g., `<<object>>` for the older representation? The `]` existed before the SDL syntax, if I'm not mistaken. 

As usual, I'm happy to see PlantUML alive and progressing! Cheers!

**Reply by The-Lum** [reply]
Hi @fuhrmanator, @arnaudroques,

_For a beginning..._
I just post a PR in order to:
- #2139

Regards,
Th.

**Reply by The-Lum** [reply]
Hello PlantUML team, @fuhrmanator, @arnaudroques,

After `<<object>>` stereotype creation and after 1rst answer to:
- https://forum.plantuml.net/16558/draw-send-signal-action-accept-event-action-activity-diagram

Here are a proposal for other UML shapes:
| UML Shape Name | Plantuml Stereotype Proposal<br>(on Activity diagram)|
|--------|--------|
| ObjectNode<br><i>(already implemented)            | `<<object>>` |
| ObjectNode<br>typed by signal	                 | `<<objectSignal>>` or `<<object-signal>>` |
| AcceptEventAction<br>without TimeEvent trigger | `<<acceptEvent>>` or `<<accept-event>>` |
| AcceptEventAction<br>with TimeEvent trigger    | `<<timeEvent>>` or `<<time-event>>` |
| SendSignalAction<br>SendObjectAction<br>with signal type | `<<sendSignal>>` or `<<send-signal>>` |
| Trigger | `<<trigger>>` |




- Do you agree?
- Or have you other proposals?

Regards,
Th.
